selftests/unit/test_utils_process.py

```diff
@@ -305,6 +306,38 @@ def handle(self, record):
                          b"should go to the log\n")
         self.assertTrue(handler.caught_record)
 
+    def test_flush_on_closed_handler(self):
+        handler = logging.StreamHandler(io.StringIO())
+        log = logging.getLogger("test_flush_on_closed_handler")
+        log.addHandler(handler)
+        read_fd, write_fd = os.pipe()
+        result = process.CmdResult()
+        fd_drainer = process.FDDrainer(read_fd, result, name="test",
+                                       stream_logger=log)
+        fd_drainer.start()
+        os.close(write_fd)
+        if fd_drainer._stream_logger is not None:
+            for handler in fd_drainer._stream_logger.handlers:
```

Comment:
Also need to ensure at least one handle has its stream closed before flush()

Reply:
Actually, this is not about trying a stream that has been closed, but one that doesn't allow for `fileno()`.

Reply:
Yes, good point.